cmd/zstreamdump/zstreamdump.c

```diff
@@ -105,6 +108,33 @@ ssread(void *buf, size_t len, zio_cksum_t *cksum)
 	return (outlen);
 }
 
+#ifdef	_UZFS
+int object_fd = -1;
+uint64_t last_object;
+
+static int
+dump_object_to_file(uint64_t object, char *buf, uint64_t len)
+{
+	char object_file[32];
+
+	if (object_fd == -1 || last_object != object) {
+		snprintf(object_file, sizeof (object_file),
+		    "%lu.dump", object);
+		if (object_fd != -1)
+			close(object_fd);
+		object_fd = open(object_file, O_RDWR|O_CREAT|O_APPEND, O_RDWR);
+		if (object_fd == -1) {
+			fprintf(stderr, "Failed to open object dump "
+			    "file '%s' err:%d\n", object_file, errno);
+			return (errno);
+		}
+		last_object = object;
+	}
+
+	return (write(object_fd, buf, len));
```

Comment:
why don't we just do printf and redirect to a file or onto a pipe?
redirecting to a file needs storage space.

Reply:
we can. but in that case, we need to execute this binary for each object.

Reply:
executing binary for every object should be fine..

Reply:
so, should we have an option to take 'object' number as input?

Reply:
actually, why do we need to execute for each object for writing to stdout?

Reply:
As of now, for checking the difference in data between two ZVOL, I need to remove zap object. That's why I have added the creation of different files as per object.
We can pass the `object` number as an argument.